src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosBlockSignatureRule.cs

```diff
@@ -86,13 +89,20 @@ private bool CheckBlockSignature(PosBlock block)
                 return false;
             }
 
-            if (ops.Count < 2) // script.GetOp(pc, opcode, vchPushValue)
+            if (ops.Count != 2)
```

Comment:
I think we can move this check even before we check `ops.ElementAt(0)`.
Also, the message `NO_SECOND_OP` is wrong here, we could have 5 Ops and think we only got 1.

Reply:
right, it should say something like `[INVALID_OP_COUNT]`

Reply:
> I think we can move this check even before we check ops.ElementAt(0).

we wanted to make super minimal changes, not beautify anything